File: src/problems_all/1672_richest_customer_wealth/1672_richest_customer_wealth.py
"""
https://leetcode.com/problems/richest-customer-wealth/

Runtime: 57 ms
Memory: 14 MB
https://leetcode.com/problems/richest-customer-wealth/submissions/879080274/
"""


from typing import List


class Solution:
    def maximumWealth(self, accounts: List[List[int]]) -> int:
        max_wealth = float('-inf')
        for account in accounts:
            total = sum(account)
            if total > max_wealth:
                max_wealth = total

        return max_wealth


def main():
    accounts = [[1, 5], [7, 3], [3, 5]]
    result = Solution().maximumWealth(accounts)
    print(result)


if __name__ == '__main__':
    main()

File: src/problems_all/1672_richest_customer_wealth/1672_richest_customer_wealth.go
/*
https://leetcode.com/problems/richest-customer-wealth/

Runtime: 5 ms
Memory: 3 MB
https://leetcode.com/problems/richest-customer-wealth/submissions/879086931/
*/

package main

import (
	"fmt"
	"math"
)

func main() {
	accounts := [][]int{{1, 2, 3}, {3, 2, 1}}
	result := maximumWealth(accounts)
	fmt.Println(result)
}

func sum(arr []int) int {
	total := 0
	for _, i := range arr {
		total += i
	}
	return total
}

func maximumWealth(accounts [][]int) int {
	maxWealth := int(math.Inf(-1))
	for _, account := range accounts {
		total := sum(account)
		if total > maxWealth {
			maxWealth = total
		}
	}
	return maxWealth
}
